### PR TITLE
Update Circleci and URL for Hek.

### LINKED ISSUE #3744: Edit on GitHub link issues
### Description

The edit on GitHub link from the docs is broken for a number of versions.
### Expected behavior

Click the link and be brought to the code on GitHub
### Actual behavior

Error as link does not exist (sometimes)
### Steps to Reproduce

Pick you favourite documentation (e.g https://docs.sunpy.org/en/stable/api/sunpy.net.fido_factory.UnifiedResponse.html) click on on the 'edit on github' link. Change the version using the version box in the bottom right of the webpage.

| doc version  | link | status |
| ------------ | ----|------- |
| latest  | https://github.com/sunpy/sunpy/tree/master/sunpy/net/fido_factory.py#L31  | ok |
| stable  | https://github.com/sunpy/sunpy/tree/1.1.0/sunpy/net/fido_factory.py#L31 | broken |
| v1.1.0 | https://github.com/sunpy/sunpy/tree/1.1.0/sunpy/net/fido_factory.py#L31 | broken |
| v1.0.7 | https://github.com/sunpy/sunpy/tree/1.0.7/sunpy/net/fido_factory.py#L31 | broken |
| v0.9.9 | http://github.com/sunpy/sunpy/tree/0.9/sunpy/net/fido_factory.py#L30 | ok |
| v0.8.5 | http://github.com/sunpy/sunpy/tree/v0.8.x/sunpy/net/fido_factory.py#L29 | broken |

For `1.x.y` the url needs to be `.../tree/v1.x.y/...` but for the the `v0.8.5` the x in the url is the issue so `.../tree/0.8.5/...` works but only without the leading  'v'?

It's not a major issue just an annoyance especially for for the 1.1.0 docs.

Seem's like this line is the issue but not sure how to fix for both 1.x.y and 0.x.y maybe add an if?
https://github.com/sunpy/sunpy/blob/fe8ef2ec505d504cb2af52d90246529e8c5fb84f/docs/conf.py#L242

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #3744 (hopefully)
Fixes https://github.com/sunpy/sunpy/issues/3859 (hopefully)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 no-backports: &no-backports
-  name: Skip any branches called backport*
+  name: Skip any branches called cherry-pick
   command: |
-    if [[ "${CIRCLE_BRANCH}" = *"backport"* ]]; then
+    if [[ "${CIRCLE_BRANCH}" == *"cherry-pick"* ]]; then
```
Comment (Cadair) [inline .circleci/config.yml]: we need both right?

Comment (nabobalis) [reply .circleci/config.yml]: Fixed.

**Comment by nabobalis** [comment]
Windows: 
```
E           hypothesis.errors.Flaky: Hypothesis test_rotatedsun_transforms(frame=HeliographicCarrington, lon=<Longitude 137.3737196 deg>, lat=<Latitude 68.6868598 deg>, obstime=<Time object: scale='utc' format='isot' value=1964-05-30T15:40:45.862>, rotated_time1=<Time object: scale='utc' format='isot' value=2004-08-09T05:00:35.804>, rotated_time2=<Time object: scale='utc' format='isot' value=1960-03-05T21:29:08.618>) produces unreliable results: Falsified on the first call but did not on a subsequent one
```

**Comment by ayshih** [comment]
Hmm, that's not a "real" failure; the build is just running much slower for whatever reason.  Given that it usually passes the deadline criterion, I suggest ignoring the failure here.

**Comment by nabobalis** [comment]
Yeah I was planning to ignore it.

**Pull request merged**